### PR TITLE
Add fugu

### PR DESCRIPTION
**Pull request by shafy**
Added a bunch of Fugu events and some smaller things that I noticed:
- In the dashboard, the div around the "Add location" button was preventing users from clicking on location markers that were "behind" that div. I've now removed that div, so it's just a button positioned in the middle
- Updated Stimulus and Turbo
- Removed UJS (not needed with Turbo and can cause conflicts)
- Needed to disable Turbo for the Dashboard header links, because it would cause rendering (and Fugu event tracking) twice (I think somehow related to caching, Turbo is always a mistery to me haha)

I'm going to add the Fugu env variables to our deployment, so this is ready to deploy!

### CODE REVIEW

**Comment by sagzy** [comment]
Thanks dude! 

- Turbo / Stimulus upgrade + using importmap: great! Thank you. As a side note, funny how many parts of the code have Turbo disabled by now :D
- Fugu: code looks good and the tracking strategy too. I have one concern though: Fugu or event tracking in general is secondary to the code logic and also not something we should have I think enabled by default, as they won't make sense  for self-hosters. One solution to this concern could be a concern (hoho), e.g.:
```
module Trackable
  extend ActiveSupport::Concern
  
  ...
  def track_event
    # here have a generic event name, as in "Viewed / Updated / Created / Deleted" followed by the resource name
  end
```

and then in the controllers only have the before_action / after_action hooks, calling this one method. If I read it correctly, all the events we are tracking so far can be done this way and the generic event names would work (and even add more consistency).

And then later on, we can easily add a constant in the concern to enable / disable tracking. e.g.:

```
module Trackable
  extend ActiveSupport::Concern
  
  EVENT_TRACKING_ENABLED = ENV.fetch("EVENT_TRACKING_ENABLED", false)
  
  ...
```

What do you think?

**Comment by shafy** [comment]
Thanks for the comments. By the way, didn't switch to importmaps yet (this is still running on webpacker), but it's coming with my next PR which is almost done.

So, how would it work then? Would evey event be called "Viewed xyz"? E.g., "Viewed DashboardController>LocationController show"? And how would we differentiate between the current event names, e.g. "Viewed Add Location" and "Added location"? AFAIK there is no good way to pass an argument to the `after_action` method, or is there?

I like the global enabling/disabling of analytics, but we could still do that by e.g. adding the conditional logic to the `FuguWorker`? It would still allow us to keep the flexibility with better event names?

**Comment by shafy** [comment]
I've looked into it, and I think we can do it something like this: Have a Trackable concern similar to what you suggested, but take in a dynamic string to set the event name (and properties optionally):

```ruby
# Trackable.rb
module Trackable
  extend ActiveSupport::Concern
  
  def track_event(name = "", properties = {})
    return unless ENV["FUGU_API_KEY] && ENV["FUGU_URL]
    # default to this, not sure the concern has access to these vars though, need to check
    name = "Viewed #{controller_name} #{action_name}" if name.empty?
    FuguWorker.perform_async(name, properties)
  end
end
```

And then in the controller do this:
```ruby
class LocationController < ApplicationController
  ...

  before_action -> { track_event("Viewed Add Location") }, only: %i[new]
  before_action -> { track_event("Added Location") }, only: %i[create]

  ...
end
```

What do you think?

**Comment by sagzy** [comment]
> I've looked into it, and I think we can do it something like this: Have a Trackable concern similar to what you suggested, but take in a dynamic string to set the event name (and properties optionally):
> 
> ```ruby
> # Trackable.rb
> module Trackable
>   extend ActiveSupport::Concern
>   
>   def track_event(name = "", properties = {})
>     return unless ENV["FUGU_API_KEY] && ENV["FUGU_URL]
>     # default to this, not sure the concern has access to these vars though, need to check
>     name = "Viewed #{controller_name} #{action_name}" if name.empty?
>     FuguWorker.perform_async(name, properties)
>   end
> end
> ```
> 
> And then in the controller do this:
> 
> ```ruby
> class LocationController < ApplicationController
>   ...
>   
>   include Trackable
>   before_action -> { track_event("Viewed Add Location") }, only: %i[new]
>   before_action -> { track_event("Added Location") }, only: %i[create]
> 
>   ...
> end
> ```
> 
> What do you think?

Yes!

**Comment by shafy** [comment]
Alrighty, done in [ae305e9](https://github.com/mapzy/mapzy/pull/49/commits/ae305e9083936be0df77a6bb09973af8db671869)

File: app/controllers/stripe_controller.rb [hunk 1]
```diff
@@ -19,6 +22,9 @@ def success_callback
     update_user_account
 
     flash[:notice] = "Subscription successful! Enjoy Mapzy!"
+
+    track_event("New Subscription", { subscription: params[:sub] })
```
Comment (sagzy) [inline app/controllers/stripe_controller.rb:26]: Couldn't these two also be written as before_action / after_action around the `checkout_session` action?

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
     after_action :remove_notice, only: %i[create]
+    after_action -> { track_event("Viewed Sign Up") }, only: %i[new]
 
     def create
       super do |resource|
         if resource.persisted?
           resource.create_account
           resource.setup_email_workers
           resource.send_welcome_email
+          track_event("New Sign Up")
```
Comment (sagzy) [inline app/controllers/users/registrations_controller.rb:18]: Couldn't we write this one as after_action for the create action?

**Review by sagzy — APPROVED** [review]
Beauty! Added 2 lil' comments

File: app/controllers/stripe_controller.rb (hunk 1 above)
Comment (shafy) [reply app/controllers/stripe_controller.rb]: Yes, good point. I wasn't sure if that would still work if e.g. there's an error, but it looks like it should not fire in that case. Fixed here: b6571fd

File: app/controllers/users/registrations_controller.rb (hunk 2 above)
Comment (shafy) [reply app/controllers/users/registrations_controller.rb]: Yes wasn't sure if I can use `resource.persisted` in the hook, but looks like it works. So here is the new commit: b6571fd

**Pull request merged**